wolfcrypt/src/port/autosar/example.c

```diff
@@ -0,0 +1,430 @@
+static int redirect_test()
+{
+    Std_ReturnType ret;
+
+    uint8 cipher[BLOCK_SIZE * 2];
+    uint32 cipherSz = 0;
+    const uint8 msg[] = { /* "Now is the time for all " w/o trailing 0 */
+        0x6e,0x6f,0x77,0x20,0x69,0x73,0x20,0x74,
+        0x68,0x65,0x20,0x74,0x69,0x6d,0x65,0x20
+    };
+    const uint8 verify[] =
+    {
+        0x95,0x94,0x92,0x57,0x5f,0x42,0x81,0x53,
+        0x2c,0xcc,0x9d,0x46,0x77,0xa2,0x33,0xcb
+    };
+    const uint8 key[] = "0123456789abcdef   ";  /* align */
+    const uint8 iv[]  = "1234567890abcdef   ";  /* align */
+    unsigned int i;
+
+    XMEMSET(cipher, 0, BLOCK_SIZE);
+
+    /* fill keystore with bad keys */
+    for (i = 0; i < MAX_KEYSTORE; i++) {
+        ret = Csm_KeyElementSet(i, CRYPTO_KE_CIPHER_KEY, verify, BLOCK_SIZE);
+        if (ret != E_OK) {
+            printf("Issue with setting key");
+            return -1;
+        }
+    }
+
+    /* set specific key that will be used for encryption */
+    ret = Csm_KeyElementSet(REDIRECTION_IN1_KEYID, REDIRECTION_IN1_KEYELMID,
+            key, BLOCK_SIZE);
+    if (ret != E_OK) {
+        printf("Issue with setting key");
+        return -1;
+    }
+
+    ret = Csm_KeyElementSet(REDIRECTION_IN2_KEYID, REDIRECTION_IN2_KEYELMID,
+            iv, BLOCK_SIZE);
+    if (ret != E_OK) {
+        printf("Issue with setting key IV");
+        return -1;
+    }
+
+    /* encrypt data using AES CBC */
+    ret = Csm_Encrypt(0U, CRYPTO_OPERATIONMODE_SINGLECALL, msg, BLOCK_SIZE,
+        cipher, &cipherSz);
+    if (ret != E_OK) {
+        printf("Issue with encrypting msg");
+        return -1;
+    }
+
+    if (XMEMCMP(cipher, verify, BLOCK_SIZE) != 0) {
+        printf("Error with cipher data ");
+        return -1;
+    }
+
+    /* now set bad key to be used for encryption */
+    ret = Csm_KeyElementSet(REDIRECTION_IN1_KEYID, REDIRECTION_IN1_KEYELMID,
+            verify, BLOCK_SIZE);
+    if (ret != E_OK) {
+        printf("Issue with setting key ");
+        return -1;
+    }
+
+    /* encrypt data using AES CBC */
+    ret = Csm_Encrypt(0U, CRYPTO_OPERATIONMODE_SINGLECALL, msg, BLOCK_SIZE,
+        cipher, &cipherSz);
+    if (ret != E_OK) {
+        printf("Issue with encrypting msg ");
+        return -1;
+    }
+
+    if (XMEMCMP(cipher, verify, BLOCK_SIZE) == 0) {
+        printf("Error with cipher data ");
+        return -1;
+    }
+
+    return 0;
+}
+#endif /* REDIRECTION_CONFIG */
+
+/* takes in test function test() and name of test
+ * returns 1 if test failed and 0 if passed */
+static int run_test(int(test)(void), const char* name)
+{
+    printf("%s", name);
+    if (test() != 0) {
+        printf("fail\n");
+        return 1;
+    }
+    else {
+        printf("pass\n");
+    }
+    return 0;
+}
+
+
+/* AES block size */
+int main(int argc, char* argv[])
+{
+    int ret = 0;
+    (void)argv;
+    (void)argc;
+
+    wolfSSL_Debugging_ON();
+    Csm_Init(NULL);
+
+    ret |= run_test(singleshot_test, "singleshot_test ... ");
```

Comment:
Don't like seeing test in an example.
Either this is an example showing different uses or this is a test program.